prover/marshal.go

```diff
@@ -180,31 +231,48 @@ func (p *DeletionParameters) UnmarshalJSON(data []byte) error {
 }
 
 type ProofJSON struct {
-	Ar  [2]string    `json:"ar"`
-	Bs  [2][2]string `json:"bs"`
-	Krs [2]string    `json:"krs"`
+	Ar            [2]string    `json:"ar"`
+	Bs            [2][2]string `json:"bs"`
+	Krs           [2]string    `json:"krs"`
+	Commitments   [][2]string  `json:"commitments"`
+	CommitmentPok [2]string    `json:"commitmentPok"`
 }
 
 func (p *Proof) MarshalJSON() ([]byte, error) {
-	const fpSize = 32
+	const fpSize = fp.Bytes
+
 	var buf bytes.Buffer
 	_, err := p.Proof.WriteRawTo(&buf)
 	if err != nil {
 		return nil, err
 	}
 	proofBytes := buf.Bytes()
+
 	proofJson := ProofJSON{}
-	proofHexNumbers := [8]string{}
-	for i := 0; i < 8; i++ {
-		proofHexNumbers[i] = toHex(new(big.Int).SetBytes(proofBytes[i*fpSize : (i+1)*fpSize]))
+	proofJson.Ar[0] = hex.EncodeToString(proofBytes[:fpSize])
```

Comment:
I wonder if you can do it better now – previously we went through bytes because the proof struct was private, but I think it is public now and we can just access the fields directly? Please check this